src/EnergyPlus/UserDefinedComponents.cc

```diff
@@ -1137,16 +1137,11 @@ namespace UserDefinedComponents {
                     }
 
                     if (!lAlphaFieldBlanks(8)) {
-                        {
-                            auto const SELECT_CASE_var(cAlphaArgs(8));
-
-                            if (SELECT_CASE_var == "YES") {
-                                state.dataUserDefinedComponents->UserCoil(CompLoop).PlantIsConnected = true;
-                            } else if (SELECT_CASE_var == "NO") {
-                                state.dataUserDefinedComponents->UserCoil(CompLoop).PlantIsConnected = false;
-                            }
+                        if (cAlphaArgs(8) == "YES") {
```

Comment:
:pushpin: There should probably be a function for this like `bool getBoolValue(std::string_view str)`.